### DISCUSSION TITLE
Raspberry Pi  / Octoprint

### DISCUSSION THREAD

**Discussion by AndyDHill** [arkypita/LaserGRBL#2014] [General]
I have a 3d Printer (bear with me)   and I use Cura to print to the printer via a raspberry PI running Octoprint which is connected via wifi. 

Octoprint works as a Wifi print server for the 3d Printer so I can send the File and Octoprint control the printer to print it.

Can something similar be set up with LaserGRBL so my laptop can sit in House and Engraver can sit in Garage so not to smell fumes. Rather than take laptop out to gragre every time I want to print.

**Comment by Fofoie** [comment]
Hi, i use virtualhere in free version to connect the raspi usb to my computer, i can access the raspi camero trought raspberry controller on my phone and vnc my computer to the phone and access lasergrbl. Warning: it doesn't work well but i didn't find why for the moment, but half of the failures are not due to virtualhere (i will try to decrease the BAUD speed in the week, to see whatt happen, but i know that i have other problem to resolve for it to work properly)

**Comment by gangemi** [comment]
My setup:
Octopi, Ender 3, 0.5W laser.

I use Lasergrbl, but run the code through some code someone built that I can't find at the moment - I think it was posted on the Facebook group.  Basically Lasergrbl doesn't seem to put in G0's for many of the Gcode commands, and the little tool adds it.  I then add some custom headers, change the extension to .gco, and send it to Octopi.  I believe I had to remove timeout warnings on Octopi to let it run correctly, but it seems to work.

Note: I haven't played with it much, and don't have good pictures created yet, but I think that's more about dialing in settings.
